src/providers/twitch/TwitchMessageBuilder.cpp

```diff
@@ -1251,14 +1251,59 @@ void TwitchMessageBuilder::appendTwitchBadges()
                 break;
             }
         }
+        else if (badge.startsWith("founder/"))
+        {
+            if (auto badgeEmote =
+                    this->twitchChannel->globalTwitchBadges().badge("founder",
+                                                                    "0"))
+            {
+                auto badgeInfo = this->tags.find("badge-info");
+                if (badgeInfo != this->tags.end() &&
+                    badgeInfo.value().toString().split(',')[0].startsWith(
+                        "founder/"))
+                {
+                    auto subMonths =
+                        badgeInfo.value().toString().split(',')[0].mid(8);
+                    this->emplace<BadgeElement>(
+                            badgeEmote.get(),
+                            MessageElementFlag::BadgeSubscription)
+                        ->setTooltip(QString((*badgeEmote)->tooltip.string) +
+                                     " (" + subMonths + " months)");
+                }
+                else
+                {
+                    this->emplace<BadgeElement>(
+                            badgeEmote.get(),
+                            MessageElementFlag::BadgeSubscription)
+                        ->setTooltip((*badgeEmote)->tooltip.string);
```

Comment:
We emplace the BadgeElement in any case, so we can move this out of a condition and avoid repeating the code.
Something like this.
```
auto badgeInfo = this->tags.find("badge-info");
QString tooltipText;
if (badgeInfo != this->tags.end() &&
    badgeInfo.value().toString().split(',')[0].startsWith(
        "founder/"))
{
    auto subMonths =
        badgeInfo.value().toString().split(',')[0].mid(8);
    tooltipText = QString((*badgeEmote)->tooltip.string) +
        " (" + subMonths + " months)";
}
else
{
    tooltipText = (*badgeEmote)->tooltip.string;
}
this->emplace<BadgeElement>(
        badgeEmote.get(),
        MessageElementFlag::BadgeSubscription)
    ->setTooltip(tooltipText);
```

The same thoughts about a `subscriber` badge emplacement.